src/test/java/com/microsoft/sqlserver/testframework/AbstractTest.java

```diff
@@ -84,10 +85,34 @@ public static void setup() throws Exception {
         applicationKey = getConfiguredProperty("applicationKey");
         keyIDs = getConfiguredProperty("keyID", "").split(Constants.SEMI_COLON);
         connectionString = getConfiguredProperty(Constants.MSSQL_JDBC_TEST_CONNECTION_PROPERTIES);
+        connectionStringNTLM = connectionString;
 
-        ds = updateDataSource(new SQLServerDataSource());
-        dsXA = updateDataSource(new SQLServerXADataSource());
-        dsPool = updateDataSource(new SQLServerConnectionPoolDataSource());
+        // if these properties are defined then NTLM is desired, modify connection string accordingly
+        String domain = System.getProperty("domain");
+        String user = System.getProperty("userNTLM");
+        String password = System.getProperty("passwordNTLM");
+
+        if (null != domain) {
+            connectionStringNTLM = TestUtils.addOrOverrideProperty(connectionStringNTLM, "domain", domain);
+        }
+
+        if (null != user) {
+            connectionStringNTLM = TestUtils.addOrOverrideProperty(connectionStringNTLM, "user", user);
+        }
+
+        if (null != password) {
+            connectionStringNTLM = TestUtils.addOrOverrideProperty(connectionStringNTLM, "password", password);
+        }
+
+        if (null != domain && null != user && null != password) {
```

Comment:
`domain` property is optional, please remove `null != domain` from the `if` statement.